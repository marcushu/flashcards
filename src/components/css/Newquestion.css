#newQuestionCard {
    position: relative;
    max-width: 600px;
    min-height: 350px;
    margin-top: 10px;
    margin-bottom: 10px;
    background-color: rgb(50 81 164);
    color: white;
    border-radius: 8px;
    padding: 15px;
}

.addQqandA {
    width: 100%;
    font-size: 16px;
    padding-top: 7px;
    padding-bottom: 7px;
    border-radius: 4px;
    border: none;
    background-color: white;
    outline: none;
}