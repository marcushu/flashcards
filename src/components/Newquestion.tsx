import { useState } from 'react';
import './css/Newquestion.css';


interface NewquestionProp {
  setNewQuestion: (newQuestion: string, newAnswer: string) => void
  hideMe: () => void
}

const Newquestion = ({ setNewQuestion, hideMe }: NewquestionProp) => {
  const [questionText, setquestionText] = useState("");
  const [anserText, setanserText] = useState("");

  return (
    <div id="newQuestionCard">
      <div id="cardBody">
        <h3 style={{ marginTop: "2px", marginBottom: "2px" }}>question:</h3>
        <input
          type="text"
          data-testid="newQuestion"
          className="addQqandA"
          onChange={e => setquestionText(e.target.value)} />
        <br />
        <h3 style={{ marginTop: "14px", marginBottom: "2px" }}>answer:</h3>

        <textarea className="addQqandA" name="newQuesiton"
          id="newQuestionTxt"
          data-testid="newAnswer"
          rows={10} cols={50}
          style={{ marginTop: "0px" }}
          onChange={e => setanserText(e.target.value)} />
      </div>
      <div id="cardFooter" style={{ fontSize: "18px" }}>
        <button className="questionButtons"
          onClick={() => setNewQuestion(questionText, anserText)}>
          Submit
        </button>
        &nbsp;
        <button className="questionButtons"
          onClick={hideMe}>
          Cancel
        </button>

      </div>
    </div>
  )
}

export default Newquestion;
