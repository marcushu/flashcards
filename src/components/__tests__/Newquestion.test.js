import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Newquestion from '../Newquestion';

describe("Newquestion", () => {
    const mockFunc = jest.fn();
    const mockFunc2 = jest.fn();

    it("renders", () => {
        const { asFragment } = render(
            <Newquestion 
                setNewQuestion={mockFunc}
                hideMe={mockFunc2} />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it("called the hideme function on 'cancel'", () => {
        const { getByRole } = render(
            <Newquestion 
                setNewQuestion={mockFunc}
                hideMe={mockFunc2} />
        );

        fireEvent.click(getByRole('button', {  name: /cancel/i}));
        expect(mockFunc2).toHaveBeenCalled();
    });

    it("loads text in the callback", () => {
        const { getByRole, getByTestId } = render(
            <Newquestion 
                setNewQuestion={mockFunc}
                hideMe={mockFunc2} />
        );

        fireEvent.change(getByTestId('newQuestion'), { target: {value: "new question"}});
        fireEvent.change(getByTestId('newAnswer'), { target: {value: "new answer"}});
        fireEvent.click(getByRole('button', {  name: /submit/i}));
        
        expect(mockFunc).toHaveBeenCalledWith("new question", "new answer");
    });
})