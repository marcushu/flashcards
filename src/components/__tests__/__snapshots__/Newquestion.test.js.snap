// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newquestion renders 1`] = `
<DocumentFragment>
  <div
    id="newQuestionCard"
  >
    <div
      id="cardBody"
    >
      <h3
        style="margin-top: 2px; margin-bottom: 2px;"
      >
        question:
      </h3>
      <input
        class="addQqandA"
        data-testid="newQuestion"
        type="text"
      />
      <br />
      <h3
        style="margin-top: 14px; margin-bottom: 2px;"
      >
        answer:
      </h3>
      <textarea
        class="addQqandA"
        cols="50"
        data-testid="newAnswer"
        id="newQuestionTxt"
        name="newQuesiton"
        rows="10"
        style="margin-top: 0px;"
      />
    </div>
    <div
      id="cardFooter"
      style="font-size: 18px;"
    >
      <button
        class="questionButtons"
      >
        Submit
      </button>
      Â 
      <button
        class="questionButtons"
      >
        Cancel
      </button>
    </div>
  </div>
</DocumentFragment>
`;
